### PR TITLE
Feature/recipe get ing

### PR DESCRIPTION
**Pull request by kammon10**
#### What's this PR do?
 add an estimate cost method to the Recipe class

#### Where should the reviewer start?
at the Recipe.js file near the bottom

#### How should this be manually tested?
 npm test

#### Any background context you want to provide?
no

#### What are the relevant tickets?
none

#### Screenshots (if appropriate)
na

#### Questions:
nope

### CODE REVIEW

**Review by cagallo — APPROVED** [review]
This method is so dry! Love it!

**Review by aliroemhildt — APPROVED** [review]
This looks awesome! almost done with this class!

File: src/classes/Recipe.js [hunk 1]
```diff
@@ -8,20 +8,44 @@ class Recipe {
...
+        acc.push(ingredient.id)
+      };
+      return acc;
+    }, []);  
+    return ingredientIDs;
+  };
+  
+  determineRecipeIngredients(ingredientsData) {
+    const ingredientIDs = this.getIngredientIDs();
+    const matchNamesWithID = ingredientIDs.reduce((acc, currentID) => {
```
Comment (anewb87) [inline src/classes/Recipe.js:23]: A find in a reduce!

File: src/classes/Recipe.js [hunk 2]
```diff
@@ -8,20 +8,44 @@ class Recipe {
...
+      acc.push(ingredientMatch.name);
+      return acc;
+    }, []);
 
-}
+    return matchNamesWithID;
+  };
 
+  calculateRecipeCostInDollars(ingredientsData) {
+    const totalIngredientsCost = this.ingredients.reduce((acc, currIngredient) => {
```
Comment (anewb87) [inline src/classes/Recipe.js:35]: Another one! Iterator methods inside iterator methods.

File: test/Recipe-test.js [hunk 3]
```diff
@@ -20,43 +18,39 @@ describe('Recipe', () => {
...
+    let ingredients = recipe. determineRecipeIngredients(ingredientsData)
+    // eslint-disable-next-line max-len
+    expect(ingredients).to.deep.equal(['wheat flour', 'bicarbonate of soda', 'eggs', 'sucrose', 'instant vanilla pudding', 'brown sugar', 'salt', 'fine sea salt', 'semi sweet chips', 'unsalted butter', 'vanilla']);
+  }); 
 
-  it.skip('should get the cost of its ingredients', () => {
-    expect(recipe.calculateRecipeCost).to.equal('suprise value!!')
-  })
+  it.only('should get the cost of its ingredients', () => {
+    expect(recipe.calculateRecipeCostInDollars(ingredientsData)).to.equal(177.76);
```
Comment (anewb87) [inline test/Recipe-test.js:48]: Very expensive cookies!

**Review by anewb87 — APPROVED** [review]
This method looks great, good practice of reduce!

**Pull request merged**